ap/tasks/histograms.py

```diff
@@ -94,34 +97,36 @@ def run(self):
                         weight = weight * events[Route(column).fields]
 
                 # get all viable category ids (only leaf categories)
-                cat_ids = []
-                for cat in self.config_inst.get_leaf_categories():
-                    cat_ids.append(cat.id)
-
-                # define & fill histograms
-                var_names = self.config_inst.variables.names()
-                with self.publish_step("looping over all variables in config ...."):
-                    for var_name in var_names:
-                        with self.publish_step("var: %s" % var_name):
-                            var = self.config_inst.variables.get(var_name)
-                            h_var = (
-                                hist.Hist.new
-                                .IntCat(cat_ids, name="category")
-                                .StrCategory([], name="shift", growth=True)
-                                .Var(var.bin_edges, name=var_name, label=var.get_full_x_title())
-                                .Weight()
+                leaf_cat_ids = [cat.id for cat in self.config_inst.get_leaf_categories()]
+
+                # define and fill histograms
+                for var_name in self.variables:
+                    variable_inst = self.config_inst.get_variable(var_name)
+                    with self.publish_step("var: %s" % var_name):
+                        h_var = (
+                            hist.Hist.new
+                            .IntCat([], name="process", growth=True)
+                            .IntCat(leaf_cat_ids, name="category")
```

Comment:
Why can't the categories axis grow like the others?

Reply:
Maybe Mathis can confirm, but I think there is a small runtime penalty with growth enabled. And I think he chose to use a fixed axis here since we now all leaf categories beforehand.